packages/typespec-azure-resource-manager/test/rules/no-response-body.test.ts

```diff
@@ -49,4 +77,87 @@ describe("typespec-azure-resource-manager: no response body rule", () => {
         message: `The body of 202 response should be empty.`,
       });
   });
+
+  it("emit warnings if 201 responses have no body", async () => {
+    await tester
+      .expect(
+        `
+        op walk(): CreatedResponse;
+      `,
+      )
+      .toEmitDiagnostics({
+        code: "@azure-tools/typespec-azure-resource-manager/no-response-body",
+        message: `The body of responses with success (2xx) status codes other than 202 and 204 should not be empty.`,
+      });
+  });
+
+  it("valid if a 2xx response has no body for head", async () => {
+    await tester
+      .expect(
+        `
+          @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+          @armProviderNamespace
+          namespace Microsoft.Foo;
+          
+          model Employee is TrackedResource<{}> {
+          @key("employeeName") @segment("employeeName") @path
+            name: string;
+           }
+           
+          @armResourceOperations
+          interface TestingOperations {
+            checkExistence is ArmResourceCheckExistence<Employee>;
+          }
+        `,
+      )
+      .toBeValid();
+  });
+
+  it("valid if a 2xx response has no body for delete", async () => {
+    await tester
+      .expect(
+        `
+          @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+          @armProviderNamespace
+          namespace Microsoft.Foo;
+          
+          model Employee is TrackedResource<{}> {
+            @key("employeeName")
+            @segment("employeeName")
+            @path
+            name: string;
+          }
+          
+          @armResourceOperations
+          interface TestingOperations {
+            delete is ArmResourceDeleteSync<Employee>;
+          }
+        `,
+      )
+      .toBeValid();
+  });
+
+  it("valid if a 2xx response has no body for post", async () => {
+    await tester
+      .expect(
+        `
+          @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+          @armProviderNamespace
+          namespace Microsoft.Foo;
+          
+          model Employee is TrackedResource<{}> {
+            @key("employeeName")
+            @segment("employeeName")
+            @path
+            name: string;
+          }
+          
+          @armResourceOperations
+          interface TestingOperations {
+            postEmployees is ArmProviderActionAsync<Response = Employee, Scope = SubscriptionActionScope>;
```

Comment:
Here I think we want `Response = void`  You could alternately use the ArmResourceActionSync template, with a void response type